internal/app/cache/cache.go

```diff
@@ -47,9 +47,6 @@ type Resource struct {
 type OnEvictFunc func(key string, value Resource)
 
 func NewCache(maxEntries int, onEvicted OnEvictFunc, ttl time.Duration) (Cache, error) {
-	if ttl < 0 {
```

Comment:
Did you choose to remove this check because it's checked for in the CLI/proto? Might be still worth having so each component can operate independently.

Reply:
Can you clarify why the check should remain?
I think if we're going to be opinionated about the proto validation being responsible for enforcing potential values we should be consistent and remove checks from other places in the code.

Reply:
My thinking around this was that there's no direct relationship between the proto and this constructor. i.e. Nothing preventing us from instantiating a cache outside of the orchestrator and giving it a TTL < 0 in the future, and have the error go uncaught. But since this is an internal package, I could see the argument for it not being necessary since a mistake like that isn't likely to happen. No strong opinions about this, I'll defer to you to make the final call!

Reply:
added it back